app/src/main/java/com/github/sdpcoachme/EditProfileActivity.kt

```diff
@@ -44,12 +58,23 @@ class EditProfileActivity : ComponentActivity() {
  * Composable used to display the user's profile.
  */
 @Composable
-fun Profile() {
+fun Profile(email: String, database: Database, future: CompletableFuture<UserInfo>) {
     // bind those to database
     var isEditing by remember { mutableStateOf(false) }
-    var fname by remember { mutableStateOf("Damian") }
-    var lname by remember { mutableStateOf("Kopp") }
-    var favsport by remember { mutableStateOf("Jogging") }
+    var fname by remember { mutableStateOf("") }
+    var lname by remember { mutableStateOf("") }
+    var favsport by remember { mutableStateOf("") }
+
+    var f by remember { mutableStateOf(future)}
+
+    f.thenAccept { newUser ->
```

Comment:
I am not sure, but it seems that doing future.thenAccept { ... } directly, without creating another future as a mutable state might be easier ?